R/bootstrap.nlsfit.R

```diff
@@ -471,6 +498,24 @@ bootstrap.nlsfit <- function(fn,
 
   errors <- apply(par.boot[rr, , drop=FALSE], 2, error, na.rm = TRUE)
 
+  if(is.null(priors$param)==FALSE && is.null(priors$p)==FALSE && is.null(priors$psamples)==FALSE){
+    lp <- length(priors$p)
+    y <- head(y,-lp)
+    dy <- head(dy,-lp)
+  }
+
+  ## Calculate the energy shifts \Delta E_n in the case of n particle correlators.
+  if(np != 1){
+    Enp <- par.boot[rr, 1, drop=FALSE]
+    DeltaE <- apply(X = as.matrix(Enp - np * msamples), MARGIN = 2, FUN = mean)
+    dDeltaE <- apply(X = as.matrix(Enp - np * msamples), MARGIN = 2, FUN = error)
+    cat("\n energy shifts\n")
```

Comment:
I do not want functions to print stuff on the screen. This clutters the output, but more importantly there is no way to further work with these results. I get that this is a convenience for you, but you should just extract that into an independent function.